src/codemodder/codemodder.py

```diff
@@ -102,14 +102,17 @@ def process_file(
 
     line_exclude = file_line_patterns(file_path, cli_args.path_exclude)
     line_include = file_line_patterns(file_path, cli_args.path_include)
-    sarif_for_file = sarif.get(str(file_path)) or {}
+    findings_for_rule = results.results_for_rule_and_file(
```

Comment:
Before, we could query the results for a file x rule in O(1) amortized due to using a dict of a dict to gather the results per rule and file. Now we have to iterate over all possible results for given rule and filter those by file path.

Since this is something we do a lot (once per file) and we expect no results per file, I suggest we organize the results by file path in a dict, as before.